File: 7-reverse-Integer.py
class Solution:
    def reverse(self, x: int) -> int:
        s = 1 if x > 0 else -1
        ans = 0
        x = abs(x)

        while x:
            a = x % 10
            b = (ans * 10) + a

            if b <= (2**31) - 1:
                ans = b
            else:
                return 0

            x //= 10

        return ans * s

File: 7-reverse-integer.java
class Solution {
    public int reverse(int x) {
        int s = x > 0 ? 1 : -1;
        int ans = 0;
        x *= s;
        while (x > 0) {
            int a = x % 10;
            long b = (long) ans * 10 + a;
            if (b <= 2147483647)
                ans = (int) b;
            else
                return 0;
            x /= 10;
        }
        return ans * s;
    }
}
